### ISSUE TITLE
Extra Grid Line On Plot (PyPlot)

### ISSUE THREAD

**Issue by detroitwilly** [matplotlib/matplotlib#11758]
### Bug report

**Bug summary**

When changing the color of major gridlines in a plot, some major grid lines will occasionally have a gray, slightly transparent line placed over it.

**Code for reproduction**

```python
# Paste your code here
#
#
import numpy as np
import matplotlib.pyplot as plt

aX = np.array([4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24])			# x-axis
aY = np.array([1,4,10,20,35,56,80,104,125,140,146,140,125,104,80,56,35,20,10,4,1])	# y-axis
aY = aY / aY.sum()

plt.plot(aX, aY)														# plot curve and show

plt.xlim((0, 30))
plt.ylim(ymin = 0)
plt.grid(True, 'major', color='k')
plt.minorticks_on()
plt.grid(True, 'minor', 'y')
plt.show()
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/1453972/43102068-bb368710-8e8f-11e8-901d-b8a6e100fa99.png)

Note the major grid line at 0.06 seems to be "shaded over".  Additionally, this issue doesn't happen if `aY = aY / aY.sum()` is commented out.

```
# If applicable, paste the console output here
#
#
```

**Expected outcome**

I would expect that all major grid lines have the color specified by `plt.grid(True, 'major', color='k')`.

**Matplotlib version**

  * Operating system: Windows 7 Professional 64bit
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: 3.6.2
  * Jupyter version (if applicable): N/A
  * Other libraries: numpy

I installed python from the available .exe file.  I installed matplotlib according to the instructions on the website (with pip).

**Comment by ImportanceOfBeingErnest** [comment]
xref: https://stackoverflow.com/questions/51485714/strange-grid-behavior-with-pyplot

Note that if we set e.g. `plt.ylim(-0.014, 0.12)` all lines become gray.

This is a problem with the locators. I.e. 

```
major = plt.gca().yaxis.major.locator()
minor = plt.gca().yaxis.minor.locator()
print(set(major) & set(minor))
```

prints indeed `{0.06}`, which is the gridline for which there is a major and minor tick present.

**Comment by ImportanceOfBeingErnest** [comment]
A possible fix is in #11762

**Comment by tacaswell** [comment]
Fixed in #11762
